scripts/setup-linux.sh

```diff
@@ -2,3 +2,12 @@
 sudo apt -y update
 sudo apt-get install -y protobuf-compiler libseccomp-dev
 
+echo "setting up rust"
+sudo rustup toolchain install 1.74 --component clippy --component rustfmt --no-self-update
+sudo rustup default 1.74
+sudo rustup target add wasm32-wasi && sudo rustup target add wasm32-unknown-unknown
+
+## setup tinygo. required for building test spin app
+echo "setting up tinygo"
```

Comment:
Where is TinyGo needed? I thought Spin apps are built through Dockerfile which already has TinyGo.

Reply:
Hello @Mossaka, we need tinygo for building this app: https://github.com/spinkube/containerd-shim-spin/tree/main/images/spin/go-hello

Reply:
We are using Docker to build images and the Dockerfile in `images/spin` will install TinyGo. Am I missing anything?

Reply:
oh apologies for the confusion. 

Now that we need to run tests with "spin registry push" version as well, we push the apps in following two ways:

- `docker build && k3d import image` (was already part of `up.sh`)
- `spin build && spin registry push <artifact>`  (added to `up.sh`)

does that answer your question?

Reply:
Ah right, that makes sense. Maybe as follow-up I wonder if we could run `docker build` and then extract the component binary out and use it as the artifact for `spin registry push`. (this will reduce the amount of deps we need to install on the host machine).